### ISSUE TITLE
Comments never shown

### ISSUE THREAD

**Issue by just-Addict** [spotweb/spotweb#188]
I've got an issue with comments on spots. I installed spotweb about 2 months back and my server has a retention of up to about 1500 to 2000 days, depending on newsgroup. When I let it retrieve spots for that far back I get comment counts for old spots but every spot younger than about 2years and 10 months shows a count of zero. 

Spots with a count bigger than zero however also never show any comments with a spot but always shows the message **No (verified) comments found.**

Checking the database with phpmyadmin I can see the **commentsxover** table **does** contain records.
However, the **usenetstate** table shows a value of zero for the **lastretrieved** field, yet the **curarticlenr** and **curmessageid** fields are both containing a value.

In short, comments apparently were downloaded during the first time the db was populated but have never shown. And seemingly don't get downloaded anymore since the **curarticlenr** field never updates.

**Comment by Sweepr** [comment]
You should never switch usenetserver's on an existing install, if for some reason you switch usenetserver in between this could potentially messup the artcilecount in your DB which often leads to truncating table's and starting from scratch.

Could you post the output of retrieve.php?

**Comment by just-Addict** [comment]
I did a fresh install when I started using this usenetserver. In fact I've done several fresh installs, all with the same result. Never any comments showing. 

Output of retrieve.php? Uh, not quite understanding this. That would be the entire spotlist, no?

**Comment by Sweepr** [comment]
Just the output that retrieve.php is showing when you run it manually from the commandline.

**Comment by just-Addict** [comment]
Ok, here's the corresponding parts of the output from retrieve.php

I've left out the parts of the posts as that goes ok.

It's the comments it bombs out of every time... without any message as to why... it just dies
I started with a fresh install, and did two update (as in retrieving new posts) attempts after that.
Every time it starts retrieving comments at the first possible post, finds 2 and then nothing...until it dies without giving any reason why.

I've had to lower the retrieve increment already down to 999 (for some reason it always retrieves 1 more per increment than specified) to not have it die on the 120 second limit during post retrieval, but at least then I get an error stating so... 
--------------------------------------------------------------------------------------
Last retrieve at 10/11/16 10:11:16
Retrieving new comments from server reader.i-telligent.com...
Appr. Message count: 	17212410
First message number:	2406383
Last message number:	19618793
Current article number:	2406383

Retrieving 2406383 till 2406385, found 2 comments in 0.2 seconds
Retrieving 2406385 till 2407384, found 0 comments in 0.16 seconds
...
Retrieving 3152638 till 3153637, found 0 comments in 0.15 seconds


----------------------------------------------------------------------------------------
Last retrieve at 10/11/16 17:46:23
Retrieving new comments from server reader.i-telligent.com...
Appr. Message count: 	17213693
First message number:	2406383
Last message number:	19620076
Current article number:	2406383

Retrieving 2406383 till 2406385, found 2 comments in 0.29 seconds
Retrieving 2406385 till 2407384, found 0 comments in 0.15 seconds
...
Retrieving 2859931 till 2860930, found 0 comments in 0.16 seconds


----------------------------------------------------------------------------------------
Last retrieve at 10/11/16 22:28:26
Retrieving new comments from server reader.i-telligent.com...
Appr. Message count: 	17214797
First message number:	2406383
Last message number:	19621180
Current article number:	2406383

Retrieving 2406383 till 2406385, found 2 comments in 0.23 seconds
Retrieving 2406385 till 2407384, found 0 comments in 0.14 seconds
...
Retrieving 3131659 till 3132658, found 0 comments in 0.16 seconds

**Comment by Sweepr** [comment]
Ok, so here's my analysis:

First message number: 2406383  <- this number is the USP's first comment.
Last message number: 19621180 <- this number is the USP's last comment.
Current article number: 2406383 <- everytime you start retrieving comments here.

So everytime you start at 2406383 , which run's fine untill the 120 seconds timeout...

Is it possbile for you to disable the 120 second timeout limit during retrieve so the script i able to catch up on the 17214797 comments remaining?

My guess:  retrieve.php get's killed before it is able to store the "Current article number" in the DB and , hence why it always starts on first message number.

**Comment by just-Addict** [comment]
the thing is, I already lowered the the retrieval increment to avoid the 120 second timeout, which only got reported during the posts retrieval, never during the comments retrieval. 

I can of course change the php file(s) to set the 120 second timeout to a ridiculous high value, but that won't fix the fact it's not retrieving any comments... the whole time it retrieved comments it said _found 0 comments in 0.xx seconds_ So any timeout value change won't do me much of anything.

**Comment by Sweepr** [comment]
You said: It's the comments it bombs out of every time... without any message as to why... it just dies

I mean that's weird, and you need to figure out why.

**Comment by just-Addict** [comment]
```
SpotWeb v0.67.28.31 on PHP v5.6.23 crashed

Fatal error occured while connecting to the newsserver:
  (205) Unexpected response [response: "Timeout after 150 seconds, articles 0, bytes 0, groups 1, posts 0, postbytes 0"]


#0 Z:\XAMPP\htdocs\spotweb\NNTP\Protocol\Client.php(662): Net_NNTP_Protocol_Client->throwError('Unexpected resp...', 205, 'Timeout after 1...')
#1 Z:\XAMPP\htdocs\spotweb\NNTP\Protocol\Client.php(1817): Net_NNTP_Protocol_Client->_handleUnexpectedResponse(205)
#2 Z:\XAMPP\htdocs\spotweb\NNTP\Client.php(1000): Net_NNTP_Protocol_Client->cmdXOver('4826165-4836164')
#3 Z:\XAMPP\htdocs\spotweb\lib\services\Nntp\Services_Nntp_Engine.php(137): Net_NNTP_Client->getOverview('4826165-4836164')
#4 Z:\XAMPP\htdocs\spotweb\lib\services\Retriever\Services_Retriever_Base.php(213): Services_Nntp_Engine->getOverview(4826165, 4836164)
#5 Z:\XAMPP\htdocs\spotweb\lib\services\Retriever\Services_Retriever_Base.php(319): Services_Retriever_Base->loopTillEnd('0', '9999')
#6 Z:\XAMPP\htdocs\spotweb\retrieve.php(188): Services_Retriever_Base->perform()
#7 {main}
```

**Comment by Sweepr** [comment]
Ah, now that's a message we can work with.

Seems you get a timeout from your USP.

**Comment by Sweepr** [comment]
Are you seeing this with a different USP aswell?

**Comment by mesa57** [comment]
Which provider do you use ? I've seen a provider which send invalid 205 messages while returning no data. Normally a 205 should only follow a QUIT command.

**Comment by just-Addict** [comment]
USP is i-telligent. 
I will also try with the ISP's news server though it has a trickle speed, so that may be a while before it's done.

**Comment by mesa57** [comment]
That was not the USP with the mentioned problem.

**Comment by just-Addict** [comment]
read back... it is.... reader.i-telligent.com

**Comment by just-Addict** [comment]
ok, that's it for me... I can't get past it so I'm throwing the towel. I refuse to use my ISP's news server with it's trickle speed.

**Comment by mesa57** [comment]
Does it work with you're ISP's news server ?

**Comment by Sweepr** [comment]
@just-Addict i understand, mostly ISP's servers are crap anyway, but i don't trust i-telligent.com aswell.

Why not try a free trial at Usenetbucket or Usenetserver for example?

**Comment by mesa57** [comment]
A pitty one cannot get a free trial at i-intellegent. Otherwise I would have executed a test run.

**Comment by mesa57** [comment]
After a few rounds the Current article number: should have been updated. Spotweb remembers, even if you cancel retrieve.php

**Comment by just-Addict** [comment]
it seems to me that the problem could be highly alleviated if the current article number (i.e. the usenetstate table) is continually updated after each successful increment is retrieved instead of only once after all posts/comments are retrieved. 

In a normal hourly/daily update situation this is already often the case since the number of new posts/comments will very likely be less than the configured increment. However, on a first run, lasting several hours, if anything goes wrong for whatever reason and spotweb crashes or is otherwise not able to finish cleanly, the last retrieved article number never gets updated. Which in my case leads to a constant restarting at the first comment article and subsequent failure due to the crash on the 205 response.

**Comment by just-Addict** [comment]
Indeed it does as you said _after each successfull {max} messages processed_, in my case there were a whole lot of susccesive _found 0 comments in 0.xx seconds_ lines in each {max} increment for the first 2.2 million articles.  Which apparently does not lead to the usenetstate record getting updated with the last processed article number.  And subsequently caused retrieve.php to start from the first available comment article number again on the next attempt.

anyways, this is all water under the bridge. It seems the situation has resolved itself. Something must have changed as retrieve is currently been running for about 11 hours as I write this, and retrieved about 5 million actual articles (~7.5 mill. if counting the first 2.2 mill. without content). So keeping fingers crossed and hopefully it'll last the remaining 15 or so hours until it's done.

**Comment by mesa57** [comment]
The average USP holds 12.000.000 comments nowadays.
